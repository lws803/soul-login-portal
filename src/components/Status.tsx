import {
  Alert,
  AlertDescription,
  AlertIcon,
  AlertTitle,
} from '@chakra-ui/react';

export default function Status({ status, message }: Props) {
  return (
    <Alert
      status="success"
      variant="subtle"
      flexDirection="column"
      alignItems="center"
      justifyContent="center"
      textAlign="center"
      minHeight="200px"
    >
      <AlertIcon boxSize="40px" mr={0} />
      <AlertTitle mt={4} mb={1} fontSize="lg">
        {status}
      </AlertTitle>
      <AlertDescription maxWidth="sm">{message}</AlertDescription>
    </Alert>
  );
}

type Props = {
  status: string;
  message: string;
};
