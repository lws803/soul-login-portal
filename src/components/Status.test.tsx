import { render } from '@testing-library/react';

import Status from './Status';

describe(Status, () => {
  it('renders', () => {
    const { container } = render(<Status status="STATUS" message="MESSAGE" />);
    expect(container).toMatchSnapshot();
  });
});
