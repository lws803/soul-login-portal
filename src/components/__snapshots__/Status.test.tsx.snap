// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status renders 1`] = `
<div>
  <div
    class="chakra-alert css-uxjdv3"
    role="alert"
  >
    <span
      class="chakra-alert__icon css-nx3xxu"
    >
      <svg
        class="chakra-icon css-fi0ww0"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="chakra-alert__title css-wrynpv"
    >
      STATUS
    </div>
    <div
      class="chakra-alert__desc css-1w05py7"
    >
      MESSAGE
    </div>
  </div>
</div>
`;
